src/action.rs

```diff
@@ -16,14 +41,65 @@ pub enum HoldTapConfig {
     /// not used in the flow of typing, like escape for example. If
     /// you are annoyed by accidental tap, you can try this behavior.
     HoldOnOtherKeyPress,
-    /// If there is a release and a press of another key, the hold
+    /// If there is a press and release of another key, the hold
     /// action is activated.
     ///
     /// This behavior is interesting for fast typist: the different
     /// between hold and tap would more be based on the sequence of
     /// events than on timing. Be aware that doing the good succession
     /// of key might require some training.
     PermissiveHold,
+    /// A custom configuration. Allows the behavior to be controlled by a caller
+    /// supplied handler function.
+    ///
```

Comment:
Some text explaining the given input (event received after the Press event on the HoldTap, in order of appearance), and the return meaning (Some if the action is determined, None if more information is needed).

Reply:
Updated the docs and doc tests too. Let me know if you'd like me to reword any of it.